### PR TITLE
Support custom formatting - especially allow easy TZ support 😄 

### PR DESCRIPTION
**Pull request by alaendle**
Arises from https://github.com/co-log/co-log/issues/217. As a first commit, just the basic changes to reduce code duplication.

### CODE REVIEW

**Comment by alaendle** [comment]
Oh dear, the CI has definitely not been running for too long 😢😉

**Comment by chshersh** [comment]
@alaendle I'll update CI in a separate PR as well where I'll move `co-log` out and remove other packages. This package hasn't been updated for a while and tools become outdated too quickly unfortunately 😞

**Comment by alaendle** [comment]
@chshersh One of my weaknesses, I'm impatient 😉. Could you please once more approve the start of the workflow? My hope is that the latest ubuntu image contains the missing lib (since the ci from co-log-core seems to work). I just see it as an intellectual challenge to get this up and running 🤦‍♂️. And even with https://github.com/nektos/act it's hard to foresee how the actual github runner will behave.

**Comment by chshersh** [comment]
@alaendle I gave Write access to this repository. I believe, if you open PRs next time, you won't need other maintainers' approval to run CI 🙂

**Comment by alaendle** [comment]
Wow thanks! CI build is running, so maybe it's now time for feedback for the change and concentrate on the improvement of the intended change. Thanks again!

**Comment by alaendle** [comment]
Happy New Year @chshersh, I don't want to push, but is there any way to get ahead? Any obstacles that I could resolve?

File: co-log/src/Colog/Message.hs
Comment (chshersh) [inline co-log/src/Colog/Message.hs:507]: I'm afraid this test won't work because `ThreadId` is non-deterministic and could be different each time 😞 
But I wonder how tests are passing here on multiple CIs 🤔

**Review by chshersh — APPROVED** [review]
@alaendle The PR looks good 👍🏻 

I was currently teaching the Haskell course so I wasn't around to do any work. But I'm still planning to refactor this repository and bootstrap the `co-log` org.

File: co-log/src/Colog/Message.hs
Comment (alaendle) [reply co-log/src/Colog/Message.hs:507]: Seems `co-log` doesn't run `doctest` - I'll take a look what changes are required to add it.

Comment (alaendle) [reply co-log/src/Colog/Message.hs:507]: See https://github.com/co-log/co-log/pull/228

**Pull request merged**
